### ISSUE TITLE
Retraining model costs too much time!

### ISSUE THREAD

**Issue by AlanLu0808** [AliaksandrSiarohin/first-order-model#363]
It is said in your paper that 

> We train our network using Adam optimizer with learning rate 2e-4 and batch size 20. The model converges in
> approximately 2 days using 2 TitanX gpus for Tai-Chi-HD and VoxCeleb.

Actually in my experiment of VoxCeleb, I found it cost much longer. I can only run 10 epoch per day. So retraining with both of "vox-256.yaml" and "vox-adv-256.yaml" will cost almost one month! Is there something goes wrong with my settings?

I trained with your code and your configuration without any modification, except batch_size is set to 16 ，for larger batchsize will cause out-of-memory. 

My Environment lists as below.
GPU:Tesla V100 * 2 ,each with 32GB memory
Pytorch 1.0

**Comment by AliaksandrSiarohin** [comment]
Are you using png data format?

**Comment by AlanLu0808** [comment]
> Are you using png data format?

yes！

**Comment by AliaksandrSiarohin** [comment]
What is the naming of the files send an example and gpu utilisation?

**Comment by AlanLu0808** [comment]
> What is the naming of the files send an example and gpu utilisation?

file names:
id10001#7w0IBEWc9Qw#000993#001143
id10001#7w0IBEWc9Qw#001298#001705
id10001#7w0IBEWc9Qw#001709#001897
id10001#7w0IBEWc9Qw#002052#002158

gpu utilisation ranges from 5% to 100%

**Comment by AlanLu0808** [comment]
> What is the naming of the files send an example and gpu utilisation?

file names:
|--train
    |--id10001#7w0IBEWc9Qw#000993#001143
        |--0.png
        |--1.png
    |--id10001#7w0IBEWc9Qw#001298#001705
        |--0.png
        |--1.png
    |--id10001#7w0IBEWc9Qw#001709#001897
    |--id10001#7w0IBEWc9Qw#002052#002158

gpu utilisation ranges from 5% to 100%

**Comment by AlanLu0808** [comment]
> What is the naming of the files send an example and gpu utilisation?

file names:
|--train
|----id10001#7w0IBEWc9Qw#000993#001143
|--------0.png
|--------1.png
|----id10001#7w0IBEWc9Qw#001298#001705
|--------0.png
|--------1.png
|----id10001#7w0IBEWc9Qw#001709#001897
|----id10001#7w0IBEWc9Qw#002052#002158

gpu utilisation ranges from 5% to 100%

**Comment by AliaksandrSiarohin** [comment]
OK, naming seems correct. Did you change anything in configs, your images is 256x256?

**Comment by AlanLu0808** [comment]
> OK, naming seems correct. Did you change anything in configs, your images is 256x256?

All I changed lists as below:
frame_shape: [256, 256, 3]
  # id_sampling: True
  # pairs_list: data/vox256.csv    # I comment this 

num_epochs: 1000
num_repeats: 7
epoch_milestones: [600, 900]   # set num_repeats to 7 and train 10X epoch， then I can make checkpoint more frequently.
batch_size: 16

**Comment by AliaksandrSiarohin** [comment]
This is not a good idea to increase number of epochs, it did heavy saving after each epoch which reduce the training speed. Also did you specify --device_ids 0,1?

**Comment by AlanLu0808** [comment]
> This is not a good idea to increase number of epochs, it did heavy saving after each epoch which reduce the training speed. Also did you specify --device_ids 0,1?

yes! 

CUDA_VISIBLE_DEVICES=4,5 python -W ignore run.py --device_ids 0,1 --config config/retrain.yaml

**Comment by alievk** [comment]
@AlanLu0808 I would check a couple of things:
1. Is GPU fully utilised? If not, find the bottleneck. It could be IO speed, slow CPU-GPU transfer etc.
2. Double check the dataset size. Just `print(len(dataset))` and multiply by `num_repeat`. You should get an adequate number here.
3. Try increase batch size. For 32Gb of memory you should get something like 30 samples in batch.
